File: Data Structures & Algorithms/two-sum-less-than-k/submission-3.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        l = 0
        r = len(nums) - 1
        maxSum = -1

        while l < r:
            sum = nums[l] + nums[r]
            if sum < k:
                maxSum = max(maxSum, sum)
                l += 1
            else:
                r -= 1
        return maxSum;

File: Data Structures & Algorithms/max-consecutive-ones/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums) {
        let curr = 0;
        let maxLen = -Infinity;
        for (let r = 0; r < nums.length; r++) {
            if (nums[r] === 1) {
                curr++;
                maxLen = Math.max(maxLen, curr);
            } else {
                curr = 0;
            }
        }

        return maxLen === -Infinity ? 0 : maxLen;
    }
}
